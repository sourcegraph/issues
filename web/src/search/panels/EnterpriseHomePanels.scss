@import './PanelContainer.scss';

.enterprise-home-panels {
    margin: 4rem 1.5rem 0 1.5rem;
    max-width: 1286px;
    display: flex;
    flex-direction: column;
    width: calc(100% - 1.5rem - 1.5rem);

    &__row {
        display: flex;
        flex-direction: row;

        .panel-container {
            height: 15rem;

            &:first-child {
                margin-right: 1.5rem;
            }
        }

        .repositories-panel {
            width: 33%;
        }

        .recent-searches-panel {
            width: 67%;
        }

        .recent-files-panel {
            width: 58%;
        }

        .saved-searches-panel {
            width: 42%;
        }

        @media (max-width: $media-md) {
            flex-direction: column;

            .panel-container {
                width: 100%;
            }
        }
    }
}
