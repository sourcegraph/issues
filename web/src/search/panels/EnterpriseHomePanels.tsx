import * as React from 'react'
import { RecentFilesPanel } from './RecentFilesPanel'
import { RecentSearchesPanel } from './RecentSearchesPanel'
import { RepositoriesPanel } from './RepositoriesPanel'
import { SavedSearchesPanel } from './SavedSearchesPanel'

export const EnterpriseHomePanels: React.FunctionComponent<{}> = () => (
    <div className="enterprise-home-panels container">
        <div className="row">
            <RepositoriesPanel className="enterprise-home-panels__panel col-md-4" />
            <RecentSearchesPanel className="enterprise-home-panels__panel enterprise-home-panels__panel-recent-searches col-md-8" />
        </div>
        <div className="row">
            <RecentFilesPanel className="enterprise-home-panels__panel col-md-7" />
            <SavedSearchesPanel className="enterprise-home-panels__panel col-md-5" />
        </div>
    </div>
)
