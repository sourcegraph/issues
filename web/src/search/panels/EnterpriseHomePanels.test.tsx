import renderer from 'react-test-renderer'
import React from 'react'
import { EnterpriseHomePanels } from './EnterpriseHomePanels'

describe('EnterpriseHomePanels', () => {
    test('render', () => {
        expect(renderer.create(<EnterpriseHomePanels />)).toMatchSnapshot()
    })
})
