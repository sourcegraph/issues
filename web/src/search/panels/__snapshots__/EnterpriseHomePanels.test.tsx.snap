// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnterpriseHomePanels render 1`] = `
<div
  className="enterprise-home-panels"
>
  <div
    className="enterprise-home-panels__row"
  >
    <div
      className="repositories-panel panel-container"
    >
      <div
        className="panel-container__header"
      >
        <h3
          className="panel-container__header-text"
        >
          Repositories
        </h3>
      </div>
      <div>
        Empty
      </div>
    </div>
    <div
      className="recent-searches-panel panel-container"
    >
      <div
        className="panel-container__header"
      >
        <h3
          className="panel-container__header-text"
        >
          Recent Searches
        </h3>
      </div>
      <div>
        Loading
      </div>
    </div>
  </div>
  <div
    className="enterprise-home-panels__row"
  >
    <div
      className="recent-files-panel panel-container"
    >
      <div
        className="panel-container__header"
      >
        <h3
          className="panel-container__header-text"
        >
          Recent Files
        </h3>
      </div>
      <div>
        Content
      </div>
    </div>
    <div
      className="saved-searches-panel panel-container"
    >
      <div
        className="panel-container__header"
      >
        <h3
          className="panel-container__header-text"
        >
          Recent Searches
        </h3>
        <button
          type="button"
        >
          Action
        </button>
      </div>
      <div>
        Content
      </div>
    </div>
  </div>
</div>
`;
